task7/script.js

```diff
@@ -0,0 +1,433 @@
+        createdAt: new Date('2020-03-17T23:00:00'),
+        author: 'Mr. Snow',
+        photoLink: 'https://m2bob-forum.net/wcf/images/avatars/3e/2720-3e546be0b0701e0cb670fa2f4fcb053d4f7e1ba5.jpg',
+        hashTags: [
+            'tag'
+        ],
+        likes: [
+            'Другой пользователь'
+        ]
+    },
+    {
+        id: '16',
+        description: 'Mr. Snow',
+        createdAt: new Date('2020-03-17T23:00:00'),
+        author: 'Mr. Snow',
+        photoLink: 'https://m2bob-forum.net/wcf/images/avatars/3e/2720-3e546be0b0701e0cb670fa2f4fcb053d4f7e1ba5.jpg',
+        hashTags: [
+            'tag'
+        ],
+        likes: [
+            'Другой пользователь'
+        ]
+    },
+    {
+        id: '17',
+        description: 'test17',
+        createdAt: new Date('2020-03-17T23:00:00'),
+        author: 'Какой-то пользователь',
+        photoLink: 'https://m2bob-forum.net/wcf/images/avatars/3e/2720-3e546be0b0701e0cb670fa2f4fcb053d4f7e1ba5.jpg',
+        hashTags: [
+            'tag'
+        ],
+        likes: [
+            'Другой пользователь'
+        ]
+    },
+    {
+        id: '18',
+        description: 'test18',
+        createdAt: new Date('2020-03-17T23:00:00'),
+        author: 'Какой-то пользователь',
+        photoLink: 'https://m2bob-forum.net/wcf/images/avatars/3e/2720-3e546be0b0701e0cb670fa2f4fcb053d4f7e1ba5.jpg',
+        hashTags: [
+            'tag'
+        ],
+        likes: [
+            'Другой пользователь'
+        ]
+    },
+    {
+        id: '19',
+        description: 'test19',
+        createdAt: new Date('2020-03-17T23:00:00'),
+        author: 'Какой-то пользователь',
+        photoLink: 'https://m2bob-forum.net/wcf/images/avatars/3e/2720-3e546be0b0701e0cb670fa2f4fcb053d4f7e1ba5.jpg',
+        hashTags: [
+            'tag'
+        ],
+        likes: [
+            'Другой пользователь'
+        ]
+    },
+    {
+        id: '20',
+        description: 'test20',
+        createdAt: new Date('2020-03-17T23:00:00'),
+        author: 'Какой-то пользователь',
+        photoLink: 'https://m2bob-forum.net/wcf/images/avatars/3e/2720-3e546be0b0701e0cb670fa2f4fcb053d4f7e1ba5.jpg',
+        hashTags: [
+            'hash', 'tag'
+        ],
+        likes: [
+            'Другой пользователь'
+        ]
+    },
+];
+
+class PostCollection {
+
+    _posts;
+
+    constructor(posts) {
+        this._posts = [];
+        this.addAll(posts);
+    }
+
+    _postSchema = {
+        id: val => typeof val === 'string' && val.length > 0,
+        description: val => typeof val === 'string' && val.length < 200,
+        createdAt: val => Object.prototype.toString.call(val) === '[object Date]',
+        author: val => typeof val === 'string' && val.length > 0,
+        photoLink: val => typeof val === 'string',
+        hashTags: val => Array.isArray(val),
+        likes: val => Array.isArray(val)
+    };
+
+    validatePost(post) {
+        if(this._posts.find(p => post.id === p.id) != null) {
```

Comment:
Minor. Code style (after if, for, ... we add sapce)
`if () {}` `for () {}` `switch () {}`
Use same code style for all of this and u and me will be happy)

Reply:
And also one empty line before `return`, don't forget)